mypy/test/data/check-inference.test

```diff
@@ -1415,13 +1415,16 @@ main: note: In member "f" of class "A":
 [case testAttributePartiallyInitializedToNoneWithMissingAnnotation]
 class A:
     def f(self) -> None:
-        self.x = None # E: Need type annotation for variable
+        self.x = None
```

Comment:
The `Need type annotation for variable` error didn't go away, it just needed to be moved to the [out] section.  The only thing that changed is that we can now infer that `x` is `int` for the body of g(), so there's an additional `int not callable error`.